platform.md

```diff
@@ -276,10 +303,34 @@ All lifecycle phases:
 - MUST read `CNB_PLATFORM_API` from the execution environment and evaluate compatibility before attempting to parse other inputs (see [Platform API Compatibility](#platform-api-compatibility))
 - MUST give command line inputs precedence over other inputs
 
+#### `preparer`
+Usage:
+```
+/cnb/lifecycle/preparer \
+  [-log-level <log-level>] \
+  [-stack-id <stack-id>] \
+  [-stack-path <stack-path>]
+```
+
+##### Inputs
+| Input         | Environment Variable    | Default Value             | Description
+|---------------|-------------------------|---------------------------|----------------------
+| `<log-level>`  | `CNB_LOG_LEVEL`       | `info`                   | Log Level
+| `<stack-id>`   | `CNB_STACK_ID`        |                          | Chosen stack ID
+| `<stack-path>` | `CNB_STACK_PATH`      | `/cnb/stack.toml`        | Path to stack file (see [`stack.toml`](#stacktoml-toml)
+
+##### Outputs
+| Output             | Description
+|--------------------|----------------------------------------------
+| [exit status]      | (see Exit Code table below for values)
+| `/dev/stdout`      | Logs (info)
+| `/dev/stderr`      | Logs (warnings, errors)
+| `<stack>`          | Path to stack file (see [`stack.toml`](#stacktoml-toml)
```

Comment:
Why does `preparer` spit out a `stack.toml`? It takes one in as an input. Does it modify the `stack.toml` in any way? I think writing a `stack.toml` is fine regardless. I assume the rationale is so that detector can pick up the `stack.toml` from a platform-specified location (like in a volume).

Reply:
I was originally thinking of the input `stack.toml` as the "path to output `stack.toml` to".

The `preparer` needs a place to emit the new stuff I've added to `stack.toml` in this proposal. I suppose it could put it somewhere as (yet-another-toml-file). 

So the way I have it now, `stack.toml` is both input and output.